addons/html_builder/static/tests/overlay_buttons.test.js

```diff
@@ -223,3 +233,65 @@ test("Use the 'clone' overlay buttons", async () => {
     expect(":iframe section").toHaveCount(2);
     expect(":iframe .col-lg-5").toHaveCount(4);
 });
+
+test("applying overlay button  should wait for actions in progress", async () => {
+    class TestPlugin extends Plugin {
+        static id = "test";
+        resources = {
+            get_overlay_buttons: this.getOverlayButtons.bind(this),
+            has_overlay_options: () => true,
+        };
+
+        getOverlayButtons(target) {
+            return [
+                {
+                    class: "test_button",
+                    title: "Test",
+                    handler: () => {
+                        target.classList.add("overlayButton");
+                    },
+                },
+            ];
+        }
+    }
+    addPlugin(TestPlugin);
+    const customActionDef = new Deferred();
+    addActionOption({
+        customAction: {
+            load: () => customActionDef,
+            apply: ({ editingElement }) => {
+                editingElement.classList.add("customAction");
+            },
+        },
+    });
+    addOption({
+        selector: ".test-options-target",
+        template: xml`<BuilderButton action="'customAction'"/>`,
+    });
```

Comment:
Since it is almost a duplicate of the other test, the only difference being the resources (which should not be a problem if they are both present in both tests), maybe extract this somewhere and call it in both tests if possible ? :slightly_smiling_face:

Reply:
It's perhaps a little bit overkill to want to generalise the test. We only have 2 occurrences. If it really becomes more redundant, we can think about it. I'm not a fan of creating a new file, I think it's better to know all the behaviour of button overlays/option containter in a single test suite.